### PR TITLE
[TimeAxis] Add end-tick-mark class to first and last tick marks

### LINKED ISSUE #1575: Time axis, make outside lines togglable on and off
We should create an API point that allows us to turn off / make invisible the outside lines on time axis (requested by slate users)
screen shot 2015-02-04 at 1 49 23 pm
http://jsfiddle.net/5rsjv76g/
In this, the line to the left of 2009 and right of 2012 should be able to become invisible

### PR DESCRIPTION
**Pull request by rcchen**
Fixes #1575

### CODE REVIEW

File: src/components/axes/timeAxis.ts [hunk 1]
```diff
@@ -399,6 +399,11 @@ export module Axis {
         attr["y2"] = this.height() - (offset + this._tierHeights[index]);
       }
       d3.select(tickMarks[0][0]).attr(attr);
+
+      // Add end-tick classes to first and last tick for CSS customization purposes
+      d3.select(tickMarks[0][0]).classed("end-tick-mark", true);
+      d3.select(tickMarks[0][tickMarks.size() - 1]).classed("end-tick-mark", true);
```
Comment (jtlan) [inline src/components/axes/timeAxis.ts]: Let's use `AbstractAxis.END_TICK_MARK_CLASS` instead.

File: test/components/timeAxisTests.ts [hunk 2]
```diff
@@ -119,4 +119,16 @@ describe("TimeAxis", () => {
...
+
+  it("adds a class corresponding to the end-tick for the first and last ticks", () => {
+    var width = 500;
+    var svg = generateSVG(width, 100);
+    scale.domain(["2010", "2014"]);
+    axis.renderTo(svg);
+    var firstTick = d3.select(".tick-mark");
+    assert.isTrue(firstTick.classed("end-tick-mark"), "first end tick has the end-tick-mark class");
+    var lastTick = d3.select(d3.selectAll(".tick-mark")[0].pop());
+    assert.isTrue(lastTick.classed("end-tick-mark"), "last end tick has the end-tick-mark class");
```
Comment (jtlan) [inline test/components/timeAxisTests.ts]: Using the static var here will make the test less fragile if someone decides to change the CSS class name.

**Comment by crmorford** [comment]
good it look

**Comment by rcchen** [comment]
GIL more plz

**Pull request merged**